lib/renderers/cairo_renderer.h

```diff
@@ -382,18 +392,24 @@ bm_cairo_paint(struct cairo *cairo, uint32_t width, uint32_t max_height, const s
                     bm_cairo_color_from_menu_color(menu, BM_COLOR_HIGHLIGHTED_FG, &paint.fg);
                     bm_cairo_color_from_menu_color(menu, BM_COLOR_HIGHLIGHTED_BG, &paint.bg);
                 }
-            } else if (bm_menu_item_is_selected(menu, items[i])) {
+            } else if (i < count && bm_menu_item_is_selected(menu, items[i])) {
                 bm_cairo_color_from_menu_color(menu, BM_COLOR_SELECTED_FG, &paint.fg);
                 bm_cairo_color_from_menu_color(menu, BM_COLOR_SELECTED_BG, &paint.bg);
-            } else if (i % 2 == 1) {
+            } else if (i < count && i % 2 == 1) {
                 bm_cairo_color_from_menu_color(menu, BM_COLOR_ALTERNATE_FG, &paint.fg);
                 bm_cairo_color_from_menu_color(menu, BM_COLOR_ALTERNATE_BG, &paint.bg);
             } else {
                 bm_cairo_color_from_menu_color(menu, BM_COLOR_ITEM_FG, &paint.fg);
                 bm_cairo_color_from_menu_color(menu, BM_COLOR_ITEM_BG, &paint.bg);
             }
+            
+            char *line_str = "error!";
```

Comment:
Don't quite understand why this is initialized as "error!", as the if / else below will always either set it to empty or to a formed message.

Reply:
I didn't really think of that, I just thought I should have it there, but looking at it it's always going to get initialized.